### ISSUE TITLE
Quoted array expansion inside loop breaks highlighting

### ISSUE THREAD

**Issue by misaki-web** [jeff-hykin/better-shell-syntax#34]
## The code with a problem is:

```bash
#!/bin/bash

command_args=("/usr/bin/test-program" "arg1" "arg2")
posts=(1 3 4 10)

echo "Start"

for post in "${posts[@]}"; do
	"${command_args[@]}" "$post"
done

echo "End"
```

## It looks like this (theme "Default Light+"):

![screenshot-bash-highlighting-bug](https://user-images.githubusercontent.com/64830812/216805258-54f6c4b4-1c7f-4071-b160-f453aeb47871.png)

## It should look like:

If the array `command_args` is not quoted, or if it's not inside a loop, syntax highlighting is OK:

![screenshot-bash-highlighting-bug-2](https://user-images.githubusercontent.com/64830812/216805326-efed39c5-2f53-474c-bdff-fee53b28e6a2.png)

![screenshot-bash-highlighting-bug-3](https://user-images.githubusercontent.com/64830812/216805372-323cdee3-f520-4bd6-8daf-1a7cc04a6dab.png)

**Comment by jeff-hykin** [comment]
For reference, here's what it looks like on my end on `v0.3.4`

lime green = executable name
<img width="596" alt="Screen Shot 2023-02-06 at 7 48 39 AM" src="https://user-images.githubusercontent.com/17692058/216988136-516bfed5-c264-4a12-a2e4-cc1eb22bdf27.png">

So right now the last one `${command_args[@]} "$post"` isn't matched correctly.

But what you're saying sounds a little different so I want to check with you to make sure I'm understanding the issue

**Comment by misaki-web** [comment]
Here's how your example is highlighted on my end when the line inside the loop is indented with a tab:

![bash-syntax-highlighting-example](https://user-images.githubusercontent.com/64830812/216998942-91808f5b-ccd4-4c55-a506-2bf0ff6d5eb3.png)

and when it's indented with spaces:

![bash-syntax-highlighting-example-3](https://user-images.githubusercontent.com/64830812/217000899-a7893630-9cc1-4496-b72d-d64d3e867632.png)

Also, here's how the original example is highlighted when the line inside the loop is indented with spaces instead of tab:

![bash-syntax-highlighting-example-2](https://user-images.githubusercontent.com/64830812/216999454-6dcc27e4-58e6-4732-a839-9fa6e77e2770.png)

**Comment by jeff-hykin** [comment]
Not sure when it got fixed, but seems to be fixed now v1.8.5
